lib/src/boilerplate_suggestors/advanced_props_and_state_class_migrator.dart

```diff
@@ -392,6 +395,94 @@ class AdvancedPropsAndStateClassMigrator extends GeneralizingAstVisitor
     } else {
       // Patch with our newly generated concrete class
       yieldPatch(node.end, node.end, newDeclarationBuffer.toString());
+
+      // Consumed props changes don't apply to state classes
+      if (isAPropsClass(node)) {
+        // Update consumedProps to preserve behavior between legacy boilerplate
+        // (only props in the main props class are consumed by default)
+        // and the new boilerplate (all mixins are consumed by default).
+        //
+        //  1. If consumedProps is already overridden, behavior will be preserved.
+        //     Bail out.
+        //
+        //  2. If this is an abstract component, then the default set of consumed
+        //     props is empty, but if we override it then it would prevent the
+        //     concrete components from consuming their props class by default.
+        //     So, we'll leave these as-is and then add a "fix me" comment
+        //     to any components extending from an abstract component, to ensure
+        //     the consumedProps impl we're adding isn't trumping that behavior.
+        //
+        //  3. If props that were consumed by default were moved to
+        //     another mixin, stub that mixin in as consumed props, and add a
+        //     "fix me" to ensure all of those props should be consumed.
+        //
+        //  4. If a mixin will be created from the props class, consume
+        //     just those props by default.
+        //
+        //  5. Otherwise, the props class didn't contain any fields, and we'll
+        //     consume no props by default.
+
+        final componentNode = getComponentNodeInRoot(node);
+        final consumedPropsDeclarations = componentNode.members
+            .whereType<MethodDeclaration>()
+            .where((method) => method.name.name == 'consumedProps');
+
+        // [1], [2]
+        if (!isAbstract(componentNode) && consumedPropsDeclarations.isEmpty) {
+          // [2]
+          var parentClassFixmeComment = '';
+          {
+            final unprefixedSuperclassName = componentNode
+                .extendsClause?.superclass?.name?.name
+                ?.split('.')
+                ?.last;
+            final extendsFromOverReactComponentBaseClass =
+                overReactBaseComponentClasses
+                    .contains(unprefixedSuperclassName);
+
+            if (!extendsFromOverReactComponentBaseClass) {
```

Comment:
Should there be a null check for `unprefixedSuperclassName` here? Or would we still need a FIXME comment in that case?

Reply:
Good question! 

I think it would be extremely unlikely for a component class to not have a superclass and make its way in here. And if that does happen, it'll just output `null` instead of crashing, which I think is good enough